### PR TITLE
refactor(components): apply new theme to all components by default [JOB-87500]

### PR DESCRIPTION
**Pull request by chris-at-jobber**
## Motivations

As the new theme is being widely rolled out, moving retheme-scoped styling to be the only styling for components.

### Changed

- Updated snapshots
- Updated docs in one or two cases where relevant

### Removed

- Removed `:global(.jobber-retheme)` selectors and moved the respective styling onto the main un-scoped selectors
- Removed `inRetheme` checks from several TSX files

## Testing

View components without `jobber-retheme` class and verify they look the same as on `retheme-feature-branch` with the `jobber-retheme` class applied

For code review, meaningful changes are easier to get to if you apply these filters to remove generated files:
image

---

[In Atlantis we use Github's built in pull request reviews](https://help.github.com/en/articles/about-pull-request-reviews).

Random photo of Atlantis

### CODE REVIEW

File: packages/components/src/Button/Button.css [hunk 1]
```diff
@@ -6,18 +6,20 @@
...
 
   --button--duration-loading: var(--timing-loading--extended);
+
+  --button--size: 40px;
 }
 
 .button {
   display: inline-flex;
-  min-height: calc(var(--base-unit) * 2.25);
+  min-height: var(--button--size);
```
Comment (chris-at-jobber) [inline packages/components/src/Button/Button.css:15]: bye-bye mystery calcs

File: packages/components/src/Button/Button.css [hunk 2]
```diff
@@ -30,23 +32,12 @@
...
-}
-
 .button * {
   color: inherit;
   fill: inherit !important;
   /* required to over-ride very specific fill from <Icon> */
 }
 
-:global(.jobber-retheme) .button * {
-  font-weight: 600 !important;
```
Comment (chris-at-jobber) [inline packages/components/src/Button/Button.css:44]: Weight and letter-spacing now handled in the Typography component

File: packages/components/src/Button/Button.css [hunk 3]
```diff
@@ -89,13 +76,6 @@ a.button:active {
   --button--color-variation--hover: var(--color-destructive--hover);
 }
```
Comment (chris-at-jobber) [inline packages/components/src/Button/Button.css:78]: Learning and subtle now use the same underlying color

File: packages/components/src/Button/Button.css [hunk 4]
```diff
@@ -159,21 +134,17 @@ a.button:active {
...
 
-:global(.jobber-retheme) .tertiary:hover,
-:global(.jobber-retheme) .tertiary:focus-visible,
-:global(.jobber-retheme) a.tertiary:hover,
-:global(.jobber-retheme) a.tertiary:focus-visible {
+.tertiary:hover,
+.tertiary:focus-visible,
+a.tertiary:hover,
+a.tertiary:focus-visible {
```
Comment (chris-at-jobber) [inline packages/components/src/Button/Button.css:147]: @fagnanm do you want to revisit the "tertiary also have borders" decision in the scope of this? I could simplify this by grouping these selectors with the `secondary` styles defined above

File: packages/components/src/Button/Button.css [hunk 5]
```diff
@@ -248,54 +200,29 @@ a.button.disabled:hover {
 }
 
 .onlyIcon {
-  width: calc(var(--base-unit) * 2.25);
+  width: var(--button--size);
   padding: 0;
 }
 
-.onlyIcon.small {
```
Comment (chris-at-jobber) [inline packages/components/src/Button/Button.css:255]: Because we're re-setting the button size variable on the `.small` and `.large` classes, we don't need to redefine te width values here

File: packages/components/src/Button/Button.tsx [hunk 6]
```diff
@@ -188,7 +188,7 @@ function ButtonInternals({ label, icon, size = "base" }: ButtonProps) {
 function getTypeSizes(size: string) {
```
Comment (chris-at-jobber) [inline packages/components/src/Button/Button.tsx:188]: Since 2 of the three cases now return "base" do we even need this function anymore? Feels like a check on the Typography component could now suffice.

File: packages/components/src/Card/Card.css [hunk 7]
```diff
@@ -64,17 +60,9 @@
 .header {
   display: flex;
   padding: var(--card--base-padding);
+  padding-bottom: 0;
   background-color: var(--card--header-background);
```
Comment (chris-at-jobber) [inline packages/components/src/Card/Card.css:64]: We could in theory just delete this declaration and the variable for the card header background now.

File: packages/components/src/Checkbox/Checkbox.css [hunk 8]
```diff
@@ -1,12 +1,11 @@
-.checkbox {
-  --checkbox-size: var(--space-base);
+.checkBoxParent {
+  --checkbox--size: 20px;
+  --checkbox--border--checked: var(--color-interactive);
```
Comment (chris-at-jobber) [inline packages/components/src/Checkbox/Checkbox.css:3]: Setting these as variables gets rid of a ton of messy selector-nesting to manage state

File: packages/components/src/Checkbox/Checkbox.css [hunk 9]
```diff
@@ -1,12 +1,11 @@
-.checkbox {
-  --checkbox-size: var(--space-base);
+.checkBoxParent {
```
Comment (chris-at-jobber) [inline packages/components/src/Checkbox/Checkbox.css:1]: Renamed this class because reading things like `.checkbox .checkBox` was seriously hard to parse

File: packages/components/src/Checkbox/Checkbox.css [hunk 10]
```diff
@@ -1,12 +1,11 @@
...
+  --checkbox--size: 20px;
```
Comment (chris-at-jobber) [inline packages/components/src/Checkbox/Checkbox.css:2]: Bye bye mystery calcs

File: packages/components/src/Checkbox/Checkbox.css (hunk 8 above)
Comment (chris-at-jobber) [reply packages/components/src/Checkbox/Checkbox.css:3]: It doesn't necessarily show up in the diff because I didn't commit alternatives, but a lot of similar-looking states have varying levels of specificity, so I had found myself either declaring the same value multiple times for different selectors throughout the file, or disable linting and over-riding specificity warnings.

File: packages/components/src/Button/Button.css (hunk 4 above)
Comment (fagnanm) [reply packages/components/src/Button/Button.css:147]: @chris-at-jobber yes, I wrote a ticket up here for it https://jobber.atlassian.net/browse/JOB-92981

File: packages/components/src/Menu/Menu.tsx [hunk 11]
```diff
@@ -284,12 +283,12 @@ function Action({
       onClick={onClick}
       ref={actionButtonRef}
     >
-      {icon && (
-        <span className={styles.icon}>
-          <Icon name={icon} />
-        </span>
-      )}
-      <Typography element="span" size="base" textColor="text">
+      {icon && <Icon name={icon} color="interactiveSubtle" />}
```
Comment (taylorvnoj) [inline packages/components/src/Menu/Menu.tsx:286]: Do we want this color here?  What about those[ instances](https://jobber.atlassian.net/browse/JOB-87494) where we want the icon color to come through? With this set here, we won't be able to see those different icon colors in the menu.

**Review by taylorvnoj — COMMENTED** [review]
I'm about half way - just wanted to put these comments up for now and I'll continue on...

File: packages/components/src/Banner/Banner.css [hunk 12]
```diff
@@ -2,18 +2,13 @@
...
-}
-
-:global(.jobber-retheme) .banner {
   padding: calc(var(--space-base) - var(--space-smaller)) var(--space-base);
   border: var(--border-base) solid var(--color-border);
   border-radius: var(--radius-base);
+  color: var(--color-text);
   background-color: var(--color-surface);
+  gap: var(--space-small);
+  align-items: center;
```
[the remark this answers is not in the document]
Comment (chris-at-jobber) [reply packages/components/src/Banner/Banner.css:11]: That is a great catch, Banner has a lot of React-side styling happening. Will get this cleaned up!

File: packages/components/src/Menu/Menu.tsx (hunk 11 above)
Comment (chris-at-jobber) [reply packages/components/src/Menu/Menu.tsx:286]: Matt actually split that out into a separate ticket but I might as well do it here!

File: packages/components/src/Drawer/Drawer.css [hunk 13]
```diff
@@ -15,60 +11,49 @@
...
+  rather than using the breakpoint custom properties.
+  DO NOT COPY THIS - USE THE BREAKPOINTS DEFINED IN THE @jobber/design package!
   */
   @media (width >= 640px), print {
     position: relative;
     z-index: auto;
     width: var(--drawer-width);
+    border: var(--border-base) solid var(--color-border);
+    border-bottom-left-radius: var(--radius-base);
+    border-top-left-radius: var(--radius-base);
```
[the remark this answers is not in the document]
Comment (chris-at-jobber) [reply packages/components/src/Drawer/Drawer.css:30]: There is a Figma but the best "source of truth" would be to do one of...
- turn on the retheme in the `useAtlantisConfig` file on master
- find the corresponding component in Jobber

I'll double check Drawer!

**Comment by chris-at-jobber** [comment]
@taylorvnoj for your Checkbox example, this branch is correct. In some places components that had `radius-base` applied when it was 2px, the new 8px `radius-base` now looks like a circle, which we don't want. Or, more generally, some things got TOO ROUND with the new overall roundness, so we've had to dial them back (see Combobox and DataList for a few similar examples)

File: packages/components/src/Banner/Banner.css (hunk 12 above)
[the remark this answers is not in the document]
Comment (chris-at-jobber) [reply packages/components/src/Banner/Banner.css:11]: Fixed [here](https://github.com/GetJobber/atlantis/pull/1840/commits/5b9eb492051090f58a004afbe80d380a7f5012dd)

**Comment by taylorvnoj** [comment]
@chris-at-jobber ohhhh that's right. Okay thank you for clarifying!

File: packages/components/src/Drawer/Drawer.css (hunk 13 above)
[the remark this answers is not in the document]
Comment (taylorvnoj) [reply packages/components/src/Drawer/Drawer.css:30]: Just checked Jobber, this is correct. It makes sense to not have the radius on the right as it is flush with the browser on the right side

File: packages/components/src/Banner/Banner.test.tsx [hunk 14]
```diff
@@ -7,9 +7,11 @@ it("renders a success banner", () => {
   expect(container).toMatchSnapshot();
 });
 
-it("adds an error class to error banners", () => {
```
Comment (chris-at-jobber) [inline packages/components/src/Banner/Banner.test.tsx:10]: We're no longer using classnames to do any changes to banner , so replaced this test with a test that does the same thing as the other tests. It's still a snapshot, which I know we don't love, but it seemed like a reasonable way to keep a similar level of coverage.

File: packages/components/src/Banner/Banner.tsx [hunk 15]
```diff
@@ -103,17 +89,10 @@ export function Banner({
       </div>
 
       {dismissible && (
-        <button
-          type="button"
-          className={styles.closeButton}
+        <ButtonDismiss
```
Comment (chris-at-jobber) [inline packages/components/src/Banner/Banner.tsx:92]: Since icon is no longer changing color, we can use the standard ButtonDismiss here now.

File: packages/components/src/Banner/Banner.css (hunk 12 above)
[the remark this answers is not in the document]
Comment (taylorvnoj) [reply packages/components/src/Banner/Banner.css:11]: We don't have to act on it because this conversation is out of scope for this PR, but I'm just noticing that the icon in the web banner is center aligned when there's multiple lines of text, and in the mobile banner the icon is at the top.  

Screenshot 2024-04-10 at 4 32 25 PM
Screenshot 2024-04-10 at 4 32 35 PM

File: packages/components/src/Drawer/Drawer.css (hunk 13 above)
[the remark this answers is not in the document]
Comment (chris-at-jobber) [reply packages/components/src/Drawer/Drawer.css:30]: Drawer did need a tiny tweak (same update to headings as Card and Modal) - updated [here](https://github.com/GetJobber/atlantis/pull/1840/commits/7e3ae6edb36a838a104ef936872df5c23f49ffee)

File: packages/components/src/Menu/Menu.tsx (hunk 11 above)
Comment (chris-at-jobber) [reply packages/components/src/Menu/Menu.tsx:286]: image

File: packages/components/src/Button/Button.css [hunk 16]
```diff
@@ -248,54 +200,29 @@ a.button.disabled:hover {
...
 }
 
 .iconOnRight > *:first-child {
   order: 1;
+  margin-left: 0;
+  margin-right: calc(var(--space-smaller) * -1);
 }
 
-:global(.jobber-retheme) .hasIconAndLabel > *:first-child {
+.hasIconAndLabel > *:first-child {
```
[the remark this answers is not in the document]
Comment (chris-at-jobber) [reply packages/components/src/Button/Button.css]: Should be [fixed now](https://github.com/GetJobber/atlantis/pull/1840/commits/707f43450b4d2651a8635636e560d2c79ea6cfca):
image

File: packages/components/src/Menu/Menu.tsx (hunk 11 above)
Comment (chris-at-jobber) [reply packages/components/src/Menu/Menu.tsx]: https://github.com/GetJobber/atlantis/pull/1840/commits/cb6de7f15a683f2ffa70924916ea3f3b30c44aaf

File: packages/components/src/Button/Button.tsx (hunk 6 above)
Comment (chris-at-jobber) [reply packages/components/src/Button/Button.tsx:188]: I was thinking something more like changing line 180 to

`size={"large" ? "large" : "base"}`

but that doesn't really change anything and leaving this how it is does give us a better outlet for managing label size going forward, so I think I will in fact leave this alone.

**Comment by chris-at-jobber** [comment]
@taylorvnoj thanks for the review! I've either made changes or left comments explaining lack of changes on some of the more "discussion"-y points.

File: packages/components/src/Menu/Menu.tsx (hunk 11 above)
Comment (taylorvnoj) [reply packages/components/src/Menu/Menu.tsx]: Brilliant! Thank you for doing that here

File: packages/components/src/Button/Button.tsx (hunk 6 above)
Comment (taylorvnoj) [reply packages/components/src/Button/Button.tsx:188]: Perfect - sounds good to me!

**Review by taylorvnoj — APPROVED** [review]
LGTM - thanks for all this work Chris 🙏  Since this is going into a feature branch, we can do a bigger QA with a pre-release when that branch becomes a PR
